sklearn/utils/extmath.py

```diff
@@ -357,6 +361,166 @@ def randomized_svd(M, n_components, n_oversamples=10, n_iter='auto',
         return U[:, :n_components], s[:n_components], V[:n_components, :]
 
 
+def lobpcg_svd(M, n_components, n_oversamples=10, n_iter='auto',
+               transpose='auto', flip_sign=True, random_state=0,
+               tol=None, explicitNormalMatrix=None):
+    """Computes a truncated SVD using LOBPCG to accelerate the randomized SVD.
+    Compared to 'randomised', the 'lobpcg' option gives more accurate
```

Comment:
Insert a blank line before this

Reply:
Sorry, I do not understand where the blank line should be inserted exactly. The text is now consistent with other defs, e.g., `def randomized_svd` above it.

There is clearly a typo (not mine) after it, in 
`def weighted_mode(a, w, axis=0):
    """Returns an array of the weighted modal (most common) value in a` 
but I am unsure how to fix it.

Reply:
Cleaning up the tests as suggested in https://github.com/scikit-learn/scikit-learn/pull/12319#discussion_r275689606 would be great, but lots of coding, not directly related to this PR to introduce and test lobpcg for SVD. I cannot do it, sorry - it is well above my knowledge of Python. 

In https://github.com/scikit-learn/scikit-learn/pull/12319#discussion_r275689773 I have just made my test consistent with the already existing test of randomized just above, which I do not understand and thus have no confidence to edit myself. I do not know what to edit. I just want to make sure that lobpcg_svd passes the same already existing tests as randomized.